src/proto/FunctionRpc.proto

```diff
@@ -553,6 +553,13 @@ message RpcException {
 
   // Textual message describing the exception
   string message = 2;
+
+  // Type of exception if thrown by user's function code 
+  optional string remote_type = 4; 
+
```

Comment:
do we also want to add a remote_stack array?

Reply:
I think so - does [this ](https://googleapis.dev/java/google-api-grpc/latest/com/google/devtools/cloudtrace/v2/StackTrace.StackFrames.Builder.html)look like the right type?

Reply:
What would be the difference between `remote_stack` and `stack_trace`?

Reply:
remote_stack is the stack trace for just the exception thrown by a user's code, and stack_trace is the full stack trace of the RpcException, which currently doesn't include the user's exception's stack trace. Added remote_stack so that workers could perpetuate the stack trace thrown by user's code to the host.

Reply:
I'm a bit confused. Here is an example stack trace for Node.js:

> Stack: Error: oops
>     at Object.<anonymous> (/Users/erijiz/SampleApps/allTheTriggers/dist/HttpError/index.js:14:15)
>     at Generator.next (<anonymous>)
>     at /Users/erijiz/SampleApps/allTheTriggers/dist/HttpError/index.js:8:71
>     at new Promise (<anonymous>)
>     at __awaiter (/Users/erijiz/SampleApps/allTheTriggers/dist/HttpError/index.js:4:12)
>     at Object.httpTrigger [as default] (/Users/erijiz/SampleApps/allTheTriggers/dist/HttpError/index.js:13:12)
>     at p.<anonymous> (/Users/erijiz/Repos/js-worker/dist/src/worker-bundle.js:2:29095)
>     at Generator.next (<anonymous>)
>     at o (/Users/erijiz/Repos/js-worker/dist/src/worker-bundle.js:2:26973)
>     at processTicksAndRejections (internal/process/task_queues.js:95:5)

Everything with "SampleApps/allTheTriggers" in the path is from "user code" and everything with "Repos/js-worker" in the path is from the worker itself. We only get one stack trace and the rpc exception already has user code in it.

Do you have a particular language worker in mind that you are designing for, perhaps .NET Isolated? If so, do you have examples of the different stack traces you're talking about?

Reply:
Huh, it looks like Node may not have this issue then, but I've validated that this is happening in .NET, and customers are seeing it in [powershell ](https://github.com/Azure/azure-functions-dotnet-worker/issues/370#issue-842977672)  as well.

Reply:
The issue still reproduces for Node.js. However, my point was that we're already sending the correct stack (as seen in the example above) over the rpc channel with the `stack_trace` property. The host doesn't need a new `remote_stack` property, it just needs to leverage the existing `stack_trace` property.

Reply:
Yup, I think we can just reuse stack_track for all languages. If `is_user_exception` is set to true, then the dotnet worker and powershell workers will need to do some additional parsing on their end, but can still use the same property